### PR TITLE
snap: add `snap known --remote`

### PR DESCRIPTION
**Pull request by mvo5**
Followup on https://github.com/snapcore/snapd/pull/1878

This adds a command to download assertions from the store via the existing `snap known` cli interface.

### CODE REVIEW

**Comment by chipaca** [comment]
I have a problem with `--store`: you want it to mean “talk to the store” but half of the time it'll look like you're telling it to store it somewhere. I think `--remote` would work a lot better.

File: cmd/snap/cmd_known.go [hunk 1]
```diff
@@ -59,6 +63,30 @@ func init() {
...
+	// FIXME: set auth context
+	var authContext auth.AuthContext
+
+	at := asserts.Type(typeName)
+	if at == nil {
+		return nil, fmt.Errorf("cannot find assertion type %q", typeName)
+	}
+	primaryKeys := make([]string, len(at.PrimaryKey))
+	for i, k := range at.PrimaryKey {
+		primaryKeys[i] = headers[k]
```
Comment (niemeyer) [inline cmd/snap/cmd_known.go]: We need something along these lines here:
```
primaryKeys[i], ok = headers[k]
if !ok {
        return nil, fmt.Errorf("missing primary header %q to query remote assertion", k)
}
```

File: cmd/snap/cmd_known.go [hunk 2]
```diff
@@ -34,6 +36,8 @@ type cmdKnown struct {
 		AssertTypeName string   `required:"true"`
 		HeaderFilters  []string `required:"0"`
 	} `positional-args:"true" required:"true"`
+
+	Store bool `long:"store"`
```
Comment (niemeyer) [inline cmd/snap/cmd_known.go]: @chipaca has a good point about store being a verb too, giving a plausible but wrong meaning to the flag.

Let's go with `--remote` as you originally suggested.

Comment (mvo5) [inline cmd/snap/cmd_known.go]: Thanks, I changed it to --remote now.

File: cmd/snap/cmd_known.go (hunk 1 above)
Comment (mvo5) [inline cmd/snap/cmd_known.go]: Good catch, thank you! I fixed this and added a test.

**Pull request merged**